core/src/test/scala/org/apache/spark/SparkThrowableSuite.scala

```diff
@@ -222,4 +222,70 @@ class SparkThrowableSuite extends SparkFunSuite {
         assert(false)
     }
   }
+
+  test("get message in the specified format") {
+    import ErrorMessageFormat._
+    class TestQueryContext extends QueryContext {
+      override val objectName = "v1"
+      override val objectType = "VIEW"
+      override val startIndex = 2
+      override val stopIndex = -1
+      override val fragment = "1 / 0"
+    }
+    val e = new SparkArithmeticException(
```

Comment:
let's also test an exception with error class but without query context.

Reply:
I will add a test w/ a sub-class but w/o query context. Thanks.

Reply:
ah, do you mean before this PR thriftserver print error class twice?